app/Main.hs

```diff
@@ -465,6 +472,55 @@ verifyPackage db paths name = do
   let srcGlobs = map (pathToTextUnsafe . (</> ("src" </> "**" </> "*.purs")) . dirFor) dependencies
   procs "purs" ("compile" : srcGlobs) empty
 
+data BowerInfoRepo = BowerInfoRepo
+  { url :: Text
+  } deriving (Show, Eq, Generic, Aeson.FromJSON)
+
+data BowerInfo = BowerInfo
+  { bower_name         :: Text
+  , bower_repository   :: BowerInfoRepo
+  , bower_dependencies :: Map.Map Text Text
+  , bower_version      :: Text
+  } deriving (Show, Eq, Generic)
+instance Aeson.FromJSON BowerInfo where
+  parseJSON = Aeson.genericParseJSON Aeson.defaultOptions
+    { fieldLabelModifier = drop 6
+    }
+
+data BowerOutput = BowerOutput
+  { latest :: BowerInfo
+  } deriving (Show, Eq, Generic, Aeson.FromJSON)
+
+addFromBower :: String -> IO ()
+addFromBower name = do
+  let bowerProc = inproc "bower" [ "info", T.pack name, "--json", "-l=error" ] empty
```

Comment:
I think this is good as a first pass, but that we'd want to either emulate the bits of Bower that we need, or make it more obvious that having bower installed as a dependency is a requirement for this feature.

Reply:
I don't want to reimplement anything from Bower personally. I doubt parsing the registry would be anywhere near as fast as using a user's existing Bower setup and all.

Could be better documented that you need Bower though, yeah.

Reply:
The registry is super simple - it's just a case of hitting a URL like http://bower.herokuapp.com/packages/purescript-halogen, the registry doesn't actually contain any of the information about the package though, so it means fetching the `bower.json` from the location the registry returns after that... which is less simple. :wink:

I don't love the idea of depending on bower either, but I understand not wanting to reimplement half of it too, so, just thought I'd throw that info in. I did quite a bit of work around this stuff for my bower-replacement-but-still-using-the-registry project that hasn't seen the light of day yet.

Reply:
FWIW, the old registry URL is deprecated, so it would be better to use this format: https://registry.bower.io/packages/purescript-halogen

(see https://twitter.com/bower/status/918073147789889536)

Reply:
Uh yes, I actually meant to use that one, must have copied it from some old code and just didn't look at it :smile: